Modules/_io/winconsoleio.c

```diff
@@ -583,14 +583,15 @@ read_console_w(HANDLE handle, DWORD maxlen, DWORD *readlen) {
                 break;
             err = 0;
             HANDLE hInterruptEvent = _PyOS_SigintEvent();
-            if (WaitForSingleObjectEx(hInterruptEvent, 100, FALSE)
-                    == WAIT_OBJECT_0) {
+            DWORD state = WaitForSingleObjectEx(hInterruptEvent, 100, FALSE);
+            if (state == WAIT_OBJECT_0 || state == WAIT_TIMEOUT) {
```

Comment:
Considering every other possible result is a hard error (which we're currently ignoring... whoops), I don't think this is the right condition.

Probably in the `WAIT_TIMEOUT` case we want to just `continue` and not call `PyErr_CheckSignals`? We certainly shouldn't reset the event, because we haven't been given it